prefect_snowflake/credentials.py

```diff
@@ -119,3 +137,70 @@ def _validate_okta_kwargs(cls, values):
                 "`okta_endpoint` must be provided"
             )
         return values
+
+    @validator("private_key")
+    def _validate_private_key(cls, private_key):
+        """
+        Ensure a private_key looks like a PEM format certificate.
+        """
+        if private_key is None:
+            return None
+
+        assert isinstance(private_key, (SecretBytes, SecretStr))
+
+        pk = private_key.get_secret_value()
+        pk = pk.decode() if isinstance(pk, bytes) else pk
+
+        if not isinstance(pk, str):
+            raise ValueError(_INVALID_PEM_FORMAT_CERTIFICATE)
```

Comment:
Perhaps these can be custom exceptions instead; e.g.
```
class InvalidPemFormat(Exception):
    """Invalid PEM Format Certificate"""
```

Reply:
Sounds like a good idea.